File: src/giada/swing/JButton.java
package giada.swing;

import static def.dom.Globals.document;

/**
 * The javax.swing.JButton clone
 *
 * @author gianpiero.diblasi
 */
public class JButton extends AbstractButton {

  public JButton() {
    super();

    this.element = document.createElement("button");
    this.element.classList.add("jbutton");
    this.element.onclick = (event) -> this.onclick();
  }

  public void setText(String text) {
    this.element.textContent = text;
  }
}

File: transpile/giada/swing/JButton.js
/**
 * The javax.swing.JButton clone
 *
 * @author gianpiero.diblasi
 */
class JButton extends AbstractButton {

  constructor() {
    super();
    this.element = document.createElement("button");
    this.element.classList.add("jbutton");
    this.element.onclick = (event) => this.onclick();
    if (LookAndFeel.CURRENT) {
      LookAndFeel.CURRENT.styleJButton(this);
    }
  }

   setText(text) {
    this.element.textContent = text;
  }
}
